File: csharp/BankAccount/BankAccount/PersonalAccountsManager.cs
namespace Banking;

public class PersonalAccountsManager
{
    private List<BankAccount> Accounts { get; } = new List<BankAccount>();

    public PersonalAccountsManager()
    {
        // in a real system it would load bank accounts from a database
        Accounts.Add(new BankAccount(new Customer("Arthur"), 800m));
    }

    public BankAccount GetAccountForCustomer(string customerName /*, string bankManager */)
    {
        var account = Accounts.Find(a => a.AccountHolder.Name == customerName);
        if (account == null)
        {
            throw new ArgumentException("unknown customer " + customerName);
        }

        return account;
    }
}

File: typescript/src/PersonalAccountsManager.ts
import { BankAccount } from './BankAccount';

export class PersonalAccountsManager {
  private readonly accounts: BankAccount[] = [];

  constructor() {
    // in a real system it would load bank accounts from a database
    this.accounts.push(new BankAccount('Arthur', 800));
  }

  getAccountForCustomer(
    customerName: string /*, String bankManager */
  ): BankAccount {
    const customerAccount = this.accounts
      .find(a => a.accountHolder === customerName);

    if (!customerAccount) {
      throw new Error(`unknown customer ${customerName}`);
    }

    return customerAccount;
  }
}
